src/transport/smtp/client/net.rs

```diff
@@ -223,6 +223,32 @@ impl NetworkStream {
         }
     }
 
+    #[cfg(any(feature = "rustls-tls", feature = "boring-tls"))]
+    pub fn certificate_chain(&self) -> Result<Vec<Vec<u8>>, Error> {
+        match &self.inner {
+            InnerNetworkStream::Tcp(_) => Err(error::client("Connection is not encrypted")),
+            #[cfg(feature = "native-tls")]
+            InnerNetworkStream::NativeTls(_) => panic!("Unsupported"),
```

Comment:
Could https://docs.rs/native-tls/0.2.12/native_tls/struct.TlsStream.html#method.peer_certificate work here?

Reply:
That only returns the server certificate, in this case we are interested in the entire trust chain and not just the final one.